packages/compiler/src/cache.ts

```diff
@@ -0,0 +1,42 @@
+import { openDB, DBSchema, IDBPDatabase } from 'idb';
+
+import { ComponentCacheRecord } from './types';
+
+const BOS_INDEX_DB = 'bosIndexedDB';
+const BOS_INDEX_DB_VERSION = 1;
+const COMPONENT_TREES_CACHE_STORE_NAME = 'componentTreesCache';
+
+interface BOSIndexDB extends DBSchema {
+  [COMPONENT_TREES_CACHE_STORE_NAME]: {
+    key: string;
+    value: ComponentCacheRecord;
+    indexes: { by_key: string };
+  };
+}
+
+export async function initializeDB(): Promise<IDBPDatabase<BOSIndexDB>> {
```

Comment:
Can we verify that we don't need error handling around IndexedDB access? (e.g. for browsers with strict storage access permissions)

If it is a realistic concern then we can create a backlog item for it, it's not critical for this PR unless there are currently browsers that provide an easy way to deny IndexedDB access.

Reply:
Yeah, that's a valid point. I guess the best place to start - to display the status of the indexedDB access in the Inspector flags tab (before the user enables this feature).